fishtank/src/cluster.ts

```diff
@@ -27,11 +54,55 @@ export class Cluster {
     return this.backend.createNetwork(this.networkName(), { attachable: true, internal: true })
   }
 
-  async spawn(options: { name: string; image?: string }): Promise<void> {
+  async spawn(options: { name: string; image?: string; config?: NodeConfig }): Promise<void> {
     const name = this.containerName(options.name)
-    return this.backend.runDetached(options.image ?? DEFAULT_IMAGE, {
+
+    const args: string[] = []
+    const runOptions: RunOptions = {
       name,
       networks: [this.networkName()],
-    })
+    }
+
+    if (options?.config) {
+      const config = options.config
+
+      if (config.networkId) {
+        args.push(`--networkId=${config.networkId}`)
+      }
+
+      if (config.cliconfig) {
+        if (config.cliconfig.configName) {
+          if (config.cliconfig.dataDir) {
+            args.push(`--config=${config.cliconfig.configName}`)
+          } else {
+            throw new Error('Need to set datadir when config name file is provided.')
+          }
+        }
+
+        if (config.cliconfig.dataDir) {
+          const parsedPath = parse(config.cliconfig.dataDir)
+
+          const dest = join(tmpdir(), name, parsedPath.name)
```

Comment:
Consider using something like:

```
const dest = join(tmpdir(), 'fishtank', this.name, name)
```

So that:
1. the tmp filesystem is not polluted with lots of directories
2. if someone creates more than one cluster, there won't be any clashing of names